lib/vtk/analytics.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'net/http'
+require 'open-uri'
+require 'uri'
+
+module Vtk
+  # Provides command analytics to VTK team
+  class Analytics
+    CLIENT_KEY = '456f76a9dc1a235690017dab981af904'
+
+    attr_reader :name, :title, :hostname
+
+    def initialize(name:, title:, hostname: nil)
+      @name = "vtk.#{name}"
+      @title = "VTK #{title}"
+      @hostname = hostname || `hostname -f`.chomp
+    end
+
+    def log
+      return if ENV['CI'] || ENV['TEST']
+
+      Process.fork do
+        begin
```

Comment:
Can the begin end block be omitted? Maybe i'm missing something

Reply:
I guess begin/end makes sense so we can rescue from within the current context

Reply:
> Can the begin end block be omitted?

Yes and no but actually yes.

Implicit `begin` in blocks was not added until Ruby 2.5. Our goal for vtk is to support all currently supported versions of Ruby, WHICH TURNS OUT... Ruby 2.4 hasn't been supported since April.

I'll update our gemspec and rubocop settings as I was (incorrectly) assuming we needed to adhere to Ruby 2.4 syntax.

Reply:
> I guess begin/end makes sense so we can rescue from within the current context

Righto. We want to make sure the internet connection is there. If it's not an exception will be raised by `Net::HTTP` and we're here to say 🔇 shhhh, don't bother the user.